gtk/src/native_dialog.rs

```diff
@@ -0,0 +1,60 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::prelude::*;
+use crate::NativeDialog;
+use crate::ResponseType;
+use std::cell::Cell;
+use std::future::Future;
+use std::pin::Pin;
+
+pub trait NativeDialogExtManual {
+    // rustdoc-stripper-ignore-next
+    /// Shows the dialog and returns a `Future` that resolves to the
+    /// `ResponseType` on response.
+    ///
+    /// ```no_run
+    /// use gtk::prelude::*;
+    ///
+    /// # async fn run() {
+    /// let dialog = gtk::FileChooserNativeBuilder::new()
+    ///    .title("Select a File")
+    ///    .build();
+    ///
+    /// dialog.run_future().await;
+    /// dialog.hide();
+    /// println!("Selected file: {:?}", dialog.file());
```

Comment:
```suggestion
    /// println!("Selected file: {:?}", dialog.file());
    /// dialog.destroy();
```
there's no point on hiding the dialog i guess, so let's provide a more realistic example

Reply:
Yes, this should be removed!